server/controllers/PasswordResetController.js

```diff
@@ -0,0 +1,119 @@
+import Mailer from '../helpers/Mailer';
+import JwtHelper from '../helpers/JwtHelper';
+import emails from '../helpers/emailMessages';
+
+const { User } = models;
+
+/**
+  * This class contains all the methods responsible for user
+  * related operations on the app including creating and querying
+  * users from the database
+  * It is made up static methods which can be called from anywhere in the app.
+  */
+export default class PasswordResetController {
+  /**
+  * Checks if email exist and sends a reset email if email exit.
+  * @param {object} req the request object
+  * @param {object} res the response object
+  */
+  static sendResetEmail(req, res, next) {
+    const { email } = req.body;
+    // Jwt token
+    const token = JwtHelper.createToken({ email }, 1800);
+    // Create Reset Email Data
+    // This is to verify if user email exist and send reset
+    // password link
+    User.findOne({ where: { email } })
+      .then((user) => {
+        if (!user) {
+          return res.status(404).send({
+            status: 'fail',
+            errors: {
+              email: [
+                'The email you provided does not exist, please check again.',
+              ],
+            },
+          });
+        }
+        const passwordResetUrl = `${req.protocol}://${req.headers.host}/api/users/account/password/reset?tokenId=${token}`;
+        const resetEmailMessage = emails.passwordReset(email, passwordResetUrl, user.firstName);
+        const message = 'A password reset link has been sent to your email. Please check your email';
+
+        // Send User Email Method Using SendGrid
+        PasswordResetController.resetProcessEmail(req, res, resetEmailMessage, message);
+      })
+      .catch(err => next(err));
+  }
+
+  /**
+  * Checks if email exist and sends a reset email if email exit.
+  * @param {object} req the request object
+  * @param {object} res the response object
+  */
+  static verifyPasswordResetToken(req, res) {
+    // A newly generated token to update the password
+    const { email } = req.decoded;
+    const passwordUpdateToken = JwtHelper.createToken({ email }, 900);
+    return res.status(200).send({
+      status: 'success',
+      message: 'Please use the link below to change your password',
+      token: passwordUpdateToken,
+      redirectUrl: `${req.protocol}://${req.headers.host}/api/users/account/password/reset?tokenId=${passwordUpdateToken}`,
+    });
+  }
+
+  /**
+  * Update user password and sends a confirmation mail on succesful password reset.
+  * @param {object} req the request object
+  * @param {object} res the response object
+  */
+  static updateUserPassword(req, res, next) {
+    // A newly generated token to update the password
+    const { email } = req.decoded;
+    const { password } = req.body;
+    const encryptedPassword = bcrypt.hashSync(password, 10);
+
+    User.find({
+      where: {
+        email,
+      }
+    }).then((updatePassword) => {
+      if (updatePassword) {
```

Comment:
Nice job on completing this task. However, 
- On this line, You are checking if the user exists before updating the database. But no code handles when the user doesn't exist. I feel this might stall the request on that middleware without returning a response.